src/Parser/Druid/Restoration/Modules/Talents/TreeOfLife.js

```diff
@@ -35,205 +44,146 @@ class TreeOfLife extends Analyzer {
-    if (this.combatants.selected.hasBuff(SPELLS.INCARNATION_TREE_OF_LIFE_TALENT.id)) {
-      if (this.tolManualApplyTimestamp !== null && event.timestamp <= this.tolManualApplyTimestamp + TREE_OF_LIFE_DURATION) {
-        if (SPELLS.REJUVENATION.id === spellId) {
-          this.totalHealingFromRejuvenationDuringToL += amount;
-        } else if (this.hasGermination && SPELLS.REJUVENATION_GERMINATION.id === spellId) {
-          this.totalHealingFromRejuvenationDuringToL += amount;
-        } else if (SPELLS.WILD_GROWTH.id === spellId) {
-          this.totalHealingFromWildgrowthsDuringToL += amount;
-        }
-        // Get total healing during ToL
-        this.totalHealingDuringToL += amount;
-      } else {
-        if (SPELLS.REJUVENATION.id === spellId) {
-          this.totalHealingFromRejuvenationDuringToLHelmet += amount;
-        } else if (this.hasGermination && SPELLS.REJUVENATION_GERMINATION.id === spellId) {
-          this.totalHealingFromRejuvenationDuringToLHelmet += amount;
-        } else if (SPELLS.WILD_GROWTH.id === spellId) {
-          this.totalHealingFromWildgrowthsDuringToLHelmet += amount;
-        }
-        // Get total healing during ToL
-        this.totalHealingDuringToLHelmet += amount;
-      }
+    accumulator.allBoostHealing += calculateEffectiveHealing(event, ALL_BOOST);
+
+    if(spellId === SPELLS.REJUVENATION.id || spellId === SPELLS.REJUVENATION_GERMINATION.id) {
+      accumulator.rejuvBoostHealing += (calculateEffectiveHealing(event, REJUV_BOOST) / ALL_MULT);
+    } else if (spellId === SPELLS.WILD_GROWTH.id) {
+      accumulator.extraWgHealing += (calculateEffectiveHealing(event, WG_INCREASE) / ALL_MULT);
     }
   }
 
   on_byPlayer_cast(event) {
     const spellId = event.ability.guid;
-    if (SPELLS.REJUVENATION.id === spellId) {
-      this.totalRejuvenationsEncounter += 1;
-    }
-
-    if (this.combatants.selected.hasBuff(SPELLS.INCARNATION_TREE_OF_LIFE_TALENT.id)) {
-      if (this.tolManualApplyTimestamp !== null && event.timestamp <= this.tolManualApplyTimestamp + TREE_OF_LIFE_DURATION) {
-        if (SPELLS.REJUVENATION.id === spellId) {
-          this.totalRejuvenationsDuringToL += 1;
-        }
-      } else {
-        if (SPELLS.REJUVENATION.id === spellId) {
-          this.totalRejuvenationsDuringToLHelmet += 1;
-        }
+    if (spellId === SPELLS.INCARNATION_TREE_OF_LIFE_TALENT.id) {
+      this.lastTolCast = event.timestamp;
+      return;
+    } else if (spellId === SPELLS.REJUVENATION.id) {
+      const accumulator = this._getAccumulator(event);
+      if (!accumulator) {
+        return;
       }
-    }
-
-    if (SPELLS.INCARNATION_TREE_OF_LIFE_TALENT.id === spellId && (this.tolManualApplyTimestamp === null || (this.tolManualApplyTimestamp + TREE_OF_LIFE_COOLDOWN) < event.timestamp)) {
-      this.tolManualApplyTimestamp = event.timestamp;
-      this.tolCasts += 1;
-      this.proccs -= 1;
+      accumulator.rejuvManaSaved += REJUV_MANA_SAVED;
```

Comment:
Should have reductions to this variable if the user didn't make use of the extra mana saved. We have these kind of calculations on promises and carafe module.

Reply:
I'll take a look